File: rgctl-tests/ecommerce-java/src/main/java/com/example/ecommerce/coolstore/rest/ProductEndpoint.java
package com.example.ecommerce.coolstore.rest;

import com.example.ecommerce.coolstore.model.CatalogProduct;
import com.example.ecommerce.coolstore.service.CoolstoreProductService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/services/products")
public class ProductEndpoint {

    private final CoolstoreProductService productService;

    public ProductEndpoint(CoolstoreProductService productService) {
        this.productService = productService;
    }

    @GetMapping
    public List<CatalogProduct> listAll() {
        return productService.getProducts();
    }

    @GetMapping("/{itemId}")
    public CatalogProduct getProduct(@PathVariable String itemId) {
        return productService.getProductByItemId(itemId);
    }
}

File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Coolstore/ProductEndpoint.cs
using Ecommerce.Coolstore.Models;
using Ecommerce.Coolstore.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Ecommerce.Coolstore;

[ApiController]
[AllowAnonymous]
[Route("services/products")]
public class ProductEndpoint : ControllerBase
{
    private readonly CoolstoreProductService _productService;

    public ProductEndpoint(CoolstoreProductService productService)
    {
        _productService = productService;
    }

    [HttpGet]
    public ActionResult<List<CatalogProduct>> ListAll() =>
        Ok(_productService.GetProducts());

    [HttpGet("{itemId}")]
    public ActionResult<CatalogProduct?> GetProduct(string itemId) =>
        Ok(_productService.GetProductByItemId(itemId));
}
